sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/messaging/MessageWithMetadata.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.messaging;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.BinaryData;
+
+/**
+ * An abstraction for a message containing a content type along with its data.
+ */
+@Fluent
```

Comment:
Fluent or Immutable? More of a high-level question on whether a message should be mutable once it has been created.

Reply:
It is mutable